app/models/submitCreateMovement/DocumentCertificateModel.scala

```diff
@@ -55,8 +55,8 @@ object DocumentCertificateModel extends ModelConstructorHelpers {
                   )
                 } else {
                   DocumentCertificateModel(
-                    documentType = Some(documentType),
-                    documentReference = None,
+                    documentType = Some(mandatoryPage(DocumentTypePage(idx)).code),
```

Comment:
documentType is already defined on line 47, so it's probably best to replace this and line 51 with
```suggestion
                    documentType = Some(documentType),
```
to prevent unnecessary user answers calls